web-bundle/src/main/java/com/graphhopper/resources/PtIsochroneResource.java

```diff
@@ -93,14 +99,22 @@ public Response doGet(
         double targetZ = seconds * 1000;
 
         GeometryFactory geometryFactory = new GeometryFactory();
-        BooleanEncodedValue accessEnc = encodingManager.getBooleanEncodedValue(VehicleAccess.key("foot"));
-        DecimalEncodedValue speedEnc = encodingManager.getDecimalEncodedValue(VehicleSpeed.key("foot"));
-        final Weighting weighting = new FastestWeighting(accessEnc, speedEnc);
-        DefaultSnapFilter snapFilter = new DefaultSnapFilter(weighting, encodingManager.getBooleanEncodedValue(Subnetwork.key("foot")));
 
-        PtLocationSnapper.Result snapResult = new PtLocationSnapper(baseGraph, locationIndex, gtfsStorage).snapAll(Arrays.asList(location), Arrays.asList(snapFilter));
-        GraphExplorer graphExplorer = new GraphExplorer(snapResult.queryGraph, gtfsStorage.getPtGraph(), weighting, gtfsStorage, RealtimeFeed.empty(), reverseFlow, false, false, 5.0, reverseFlow, blockedRouteTypes);
+        BooleanEncodedValue accessProfileEnc = encodingManager.getBooleanEncodedValue(VehicleAccess.key(accessProfile));
+        DecimalEncodedValue accessSpeedEnc = encodingManager.getDecimalEncodedValue(VehicleSpeed.key(accessProfile));
+        final Weighting accessWeighting = new FastestWeighting(accessProfileEnc, accessSpeedEnc);
+        DefaultSnapFilter accessSnapFilter = new DefaultSnapFilter(accessWeighting, encodingManager.getBooleanEncodedValue(Subnetwork.key(accessProfile)));
+        BooleanEncodedValue egressProfileEnc = encodingManager.getBooleanEncodedValue(VehicleAccess.key(egressProfile));
+        DecimalEncodedValue egressSpeedEnc = encodingManager.getDecimalEncodedValue(VehicleSpeed.key(egressProfile));
+        final Weighting egressWeighting = new FastestWeighting(egressProfileEnc, egressSpeedEnc);
+        DefaultSnapFilter egressSnapFilter = new DefaultSnapFilter(egressWeighting, encodingManager.getBooleanEncodedValue(Subnetwork.key(egressProfile)));
+        PtLocationSnapper.Result snapResult = new PtLocationSnapper(baseGraph, locationIndex, gtfsStorage).snapAll(Arrays.asList(location), Arrays.asList(accessSnapFilter, egressSnapFilter));
+
+        // TODO: Should also incorporate egressWeighting in some way? Maybe?
```

Comment:
Ah. The issue is that when we exit a station while expanding our circle, we don't know yet if we are going to take another transit leg, or we are already on our egress leg. The router class doesn't support creating labels for both at once.

I think what you chose to do is fine -- only consider one street mode, and call it access or egress depending on the "direction". It does mean that when we say "car", we will take our transfers by "car" as well, so we basically take our car on the bus.

To be resolved another time.